lib/howl/interactions/line_selection.moon

```diff
@@ -49,13 +49,24 @@ line_match_highlighter = (editor) ->
             continue
           line = buffer.lines[nr]
           start_pos = line.start_pos
-          segments = Matcher.explain text, line.text
+          segments = explain text, line.text
           if segments
             for segment in *segments
               ranges[#ranges + 1] = { start_pos + segment[1] - 1, segment[2] }
 
         highlight.apply 'search_secondary', buffer, ranges
 
+create_matcher = (find) ->
+  class CustomMatcher
```

Comment:
Ah, I see. In that case I propose a more lightweight solution by returning a table with the `explain`, with a metatable set containing the `__call`.